agent/xdsv2/rbac_resources.go

```diff
@@ -169,6 +174,196 @@ func makeRBACPolicy(p *pbproxystate.Permission) *envoy_rbac_v3.Policy {
 	}
 }
 
+func makeL7RBACPolicies(l7Permissions []*pbproxystate.Permission) map[string]*envoy_rbac_v3.Policy {
+	// sort permissions into those with L7-specific features and those without, to match labeling and behavior
+	// conventions in V1: https://github.com/hashicorp/consul/blob/4e451f23584473a7eaf7f123145ca85e0a31783a/agent/xds/rbac.go#L647
+	// this is a somewhat unfortunate carry-over needed for testing v1 vs v2 final config
+	// and this will break with v1 intentions when multiple L4 permissions are used
+	var l4Perms []*pbproxystate.Permission
+	var l7Perms []*pbproxystate.Permission
+	for _, p := range l7Permissions {
+		if len(p.DestinationRules) > 0 {
+			l7Perms = append(l7Perms, p)
+		} else {
+			l4Perms = append(l4Perms, p)
+		}
+	}
+	l4PolicyLabel := func(i int) string {
+		if len(l4Perms) == 1 {
+			return baseL4PermissionKey
+		}
+		return fmt.Sprintf("%s-%d", baseL4PermissionKey, i)
+	}
+
+	policies := make(map[string]*envoy_rbac_v3.Policy, len(l7Permissions))
+
+	// L7 policies first, then L4 per: https://github.com/hashicorp/consul/blob/4e451f23584473a7eaf7f123145ca85e0a31783a/agent/xds/rbac.go#L664
```

Comment:
Is there a good reason for following this on v2? Just testing?

Reply:
Just ease of comparison testing. This can probably be changed with some work on testing utilities.

Reply:
After giving this a second look, I'm not seeing where this v1 ordering is that you are referring to is. Could you point me to it?